plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -320,7 +320,7 @@ func (w *windowsEventLog) getRecord(evtHandle EvtHandle) (*windowsEventLogRecord
 	var bufferUsed uint32
 	err = EvtFormatMessage(publisherMetadataEvtHandle, evtHandle, 0, 0, 0, EvtFormatMessageXml, uint32(bufferSize), &renderBuf[0], &bufferUsed)
 	EvtClose(publisherMetadataEvtHandle)
-	if err != nil {
+	if err != nil && bufferUsed == 0 {
 		return nil, fmt.Errorf("EvtFormatMessage() publisher %v, err %v", newRecord.System.Provider.Name, err)
 	}
```

Comment:
The test was working for Windows Server 2019.
It was failing for Windows Server 2022 with the `err`:
`The message ID for the desired message could not be found`.

It turns out that despite getting an error, `EvtFormatMessage()` was returning the expected message in the `renderBuf` slice.

In my opinion, if data is returned then the `err` is more like a _warning_.
And we should report the event log message if we can.

If desired I could add a message to CloudWatchAgent.log for this scenario:
```
if err != nil {
    log.Printf("D! [wineventlog] EvtFormatMessage() publisher %v, err %v", newRecord.System.Provider.Name, err)
    if bufferUsed == 0 {
        return nil, fmt.Errorf("EvtFormatMessage() publisher %v, err %v", newRecord.System.Provider.Name, err)
    }
}
```
Given that is will probably occur a lot I have chosen not to add it.